### PR TITLE
WIP: zabbix_action - fix error with zabbix 5

### LINKED ISSUE #92: Error in zabbix_action
##### SUMMARY
Error on zabbix_action if there's no change.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
zabbix_action

##### ANSIBLE VERSION
```paste below
ansible 2.9.9
  config file = /d/Code/gitlab/zabbix-servers/ansible/ansible.cfg
  configured module search path = ['/home/fism/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /d/Code/gitlab/zabbix-servers/venv/lib/python3.8/site-packages/ansible
  executable location = /d/Code/gitlab/zabbix-servers/venv/bin/ansible
  python version = 3.8.2 (default, Apr 27 2020, 15:53:34) [GCC 9.3.0]

community.zabbix==0.2.0
```

##### ENVIRONMENT
Zabbix 5.0.1 (centos containers)

##### STEPS TO REPRODUCE
Running the play the first time creates the action. Subsequent runs are failing.

```yaml
- name: "Defining default proxy actions"
      delegate_to: localhost
      zabbix_action:
        server_url: "{{ zabbix_apiserver_url }}"
        login_user: "{{ zabbix_apiuser_rw }}"
        login_password: "{{ zabbix_apipassword_rw }}"
        event_source: auto_registration
        name: "{{ zabbix_action_name }}"
        status: enabled
        esc_period: 1h
        state: present
        formula: A
        conditions:
          - type: 'proxy'
            operator: '='
            value: "proxya"
            formulaid: A
        operations:
          - type: 'add_to_host_group'
            host_groups: "INTRA"
```

##### EXPECTED RESULTS
Subsequent runs shouldn't fail.

##### ACTUAL RESULTS
```paste below
The full traceback is:
  File "/tmp/ansible_zabbix_action_payload_mujf4st6/ansible_zabbix_action_payload.zip/ansible/modules/zabbix_action.py", line 882, in update_action
  File "/d/Code/gitlab/zabbix-servers/venv/lib/python3.8/site-packages/zabbix_api.py", line 341, in method
    return self.universal("%s.%s" % (self.data["prefix"], name), opts[0])
  File "/d/Code/gitlab/zabbix-servers/venv/lib/python3.8/site-packages/zabbix_api.py", line 79, in wrapper
    return self.do_request(self.json_obj(method, opts))['result']
  File "/d/Code/gitlab/zabbix-servers/venv/lib/python3.8/site-packages/zabbix_api.py", line 348, in do_request
    return self.parent.do_request(req)
  File "/d/Code/gitlab/zabbix-servers/venv/lib/python3.8/site-packages/zabbix_api.py", line 299, in do_request
    raise ZabbixAPIException(msg, jobj['error']['code'])
fatal: [zabbixintra01p -> localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "acknowledge_default_message": "",
            "acknowledge_default_subject": "",
            "acknowledge_operations": [],
            "conditions": [
                {
                    "formulaid": "A",
                    "operator": "=",
                    "type": "proxy",
                    "value": "xxx",
                    "value2": null
                }
            ],
            "default_message": "",
            "default_subject": "",
            "esc_period": "1h",
            "eval_type": null,
            "event_source": "auto_registration",
            "formula": "A",
            "http_login_password": null,
            "http_login_user": null,
            "login_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "login_user": "xxx",
            "name": "AUTO - add Proxy Group - intra",
            "operations": [
                {
                    "command": null,
                    "command_type": null,
                    "esc_period": null,
                    "esc_step_from": 1,
                    "esc_step_to": 1,
                    "execute_on": null,
                    "host_groups": [
                        "INTRA"
                    ],
                    "inventory": null,
                    "media_type": null,
                    "message": null,
                    "operation_condition": null,
                    "password": null,
                    "port": null,
                    "run_on_groups": null,
                    "run_on_hosts": null,
                    "script_name": null,
                    "send_to_groups": null,
                    "send_to_users": null,
                    "ssh_auth_type": "password",
                    "ssh_privatekey_file": null,
                    "ssh_publickey_file": null,
                    "subject": null,
                    "templates": null,
                    "type": "add_to_host_group",
                    "username": null
                }
            ],
            "pause_in_maintenance": true,
            "recovery_default_message": "",
            "recovery_default_subject": "",
            "recovery_operations": [],
            "server_url": "http://xxx/",
            "state": "present",
            "status": "enabled",
            "timeout": 10,
            "validate_certs": true
        }
    },
    "msg": "Failed to update action '28': ('Error -32500: Application error., Cannot perform update statement on table \"actions\" without values. while sending {\"jsonrpc\": \"2.0\", \"method\": \"action.update\", \"params\": {\"def_longdata\": \"\", \"def_shortdata\": \"\", \"r_longdata\": \"\", \"r_shortdata\": \"\", \"ack_longdata\": \"\", \"ack_shortdata\": \"\", \"actionid\": \"28\"}, \"auth\": \"xxx\", \"id\": 6}', -32500)"
}
```

### PR DESCRIPTION
**Pull request by rockaut**
##### SUMMARY
Removes Action fields as the are removed from Zabbix 5.0 API
Fixes #92 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
zabbix_action

### CODE REVIEW

File: plugins/modules/zabbix_action.py [hunk 1]
```diff
@@ -182,25 +182,33 @@
...
     default_subject:
         description:
             - Problem message default subject.
+            - With >= Zabbix 5.0 this field is removed from the API and is dropped silently by module.
+            - Works only with < Zabbix 5.0
     recovery_default_message:
         description:
             - Recovery message text.
+            - With >= Zabbix 5.0 this field is removed from the API and is dropped silently by module.
             - Works only with >= Zabbix 3.2
```
Comment (D3DeFi) [inline plugins/modules/zabbix_action.py:196]:
```suggestion
            - Works only with >= Zabbix 3.2 and < Zabbix 5.0
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## devel
 
+### Improvements
+#### Modules:
+  - zabbix_action - fixed error on changed API fields for Zabbix 5.0
```
Comment (sky-joker) [inline CHANGELOG.md:7]: How about add at the end the related for PR or issue link?
e.g.

```
- zabbix_action - fixed error on changed API fields for Zabbix 5.0(https://github.com/ansible-collections/community.zabbix/pull/94)
```

What do you think @D3DeFi ?

Comment (rockaut) [reply CHANGELOG.md:7]: Oh I absolutely like that. Wasn't that even a standard/test/check in old repo?

Comment (sky-joker) [reply CHANGELOG.md]: Thank you for an additional link of the issue.
When the Zabbix module was in Ansible upstream(2.9 and less), we wrote it in a fragmentation file.

**Comment by sky-joker** [comment]
LGTM

**Comment by D3DeFi** [comment]
Thanks both!

**Pull request merged**